### PR TITLE
ramips: mt7621: add support for Xiaomi Mi Router 4A Gigabit Edition

### PR DESCRIPTION
**Pull request by rogerpueyo**
This commit adds support for the Xiaomi Mi Router 4A Gigabit Edition, an MT7621-based dual-band AC wireless router with 3 Gigabit Ethernet ports.

While the hardware is fully supported, it is not trivial to flash OpenWrt on it: no SSH/Telnet access, no TTL console, no way to stop the bootloader, firmware upgrade checks for RSA signatures, so does the U-Boot-based recovery process, etc. The only known procedure, so far, is to dump the SPI memory, modify it so that the bootloader can be stopped and then boot an initramfs "as usual".

There is a long thread with details about it at https://forum.openwrt.org/t/36685, where some people are reporting successful flashings.

### CODE REVIEW

File: target/linux/ramips/dts/mt7621_xiaomi_mir4a-gigabit.dts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c";
+			ralink,function = "gpio";
+		};
+	};
+};
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/mt7621_xiaomi_mir4a-gigabit.dts]: Add newline.

**Comment by adrianschmutzler** [comment]
WAN MAC address in 0xe006? Label MAC address?

File: target/linux/ramips/dts/mt7621_xiaomi_mir4a-gigabit.dts (hunk 1 above)
Comment (rogerpueyo) [reply target/linux/ramips/dts/mt7621_xiaomi_mir4a-gigabit.dts]: ACK

**Comment by rogerpueyo** [comment]
> WAN MAC address in 0xe006?

Yes, the stock firmware's LAN MAC address is at 0xe000 and the WAN MAC address at 0xe006.

>Label MAC address?

The label has the WAN MAC address at 0xe006. How to tag this?

**Comment by adrianschmutzler** [comment]
Add a case to MAC setup in 02_network:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L516

xiaomi,mir4a-gigabit)
   wan_mac=$(mtd_get_mac_binary factory 0xe006)
   label_mac=$wan_mac
   ;

**Comment by rogerpueyo** [comment]
> Add a case to MAC setup in 02_network:
> https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L516
> 
> xiaomi,mir4a-gigabit)
> wan_mac=$(mtd_get_mac_binary factory 0xe006)
> label_mac=$wan_mac
> ;

Done, thanks :)

**Comment by rogerpueyo** [comment]
Added a second commit to support the device with uboot-envtools.

**Comment by juppin** [comment]
There is already a similar pr on the mailing list which states that it support also this device...
https://patchwork.ozlabs.org/patch/1154310

**Comment by rogerpueyo** [comment]
OK then.

**Comment by juppin** [comment]
Would you mind to test the pr on the mailing list and give feedback?

**Comment by rogerpueyo** [comment]
> Would you mind to test the pr on the mailing list and give feedback?

Both PRs work the same. :+1:

**Comment by Neustradamus** [comment]
Merged commit is:
- https://github.com/openwrt/openwrt/commit/a51e46e543ef2e923fb97fb092d04355dfca7d7d

**Pull request closed without merging**